tests/FakeItEasy.Analyzer.Tests.Shared/Helpers/ArgumentConstraintTestCases.cs

```diff
@@ -0,0 +1,19 @@
+namespace FakeItEasy.Analyzer.Tests.Helpers
+{
+    using Xunit;
+
+    public static class ArgumentConstraintTestCases
+    {
+        public static TheoryData<string> From(params string[] constraints)
+        {
+            var theoryData = new TheoryData<string>();
+            foreach (var constraint in constraints)
+            {
+                theoryData.Add(constraint);
+                theoryData.Add("An" + constraint.Substring(1));

```

Comment:
Clever. I was about to remark on the missing `An` cases, but it looks like you have it covered.
My only worry is that it's not obvious that the input is supposed to start with `A.`... Maybe throw if it doesn't, so that we don't forget and get weird results when we pass something else? Or maybe just a comment would be enough.